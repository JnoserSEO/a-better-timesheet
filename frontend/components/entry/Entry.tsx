import React from 'react'
import '../../styles/entry/entry.scss'

export interface EntryProps {
    hours:number;
    name:string;
}

const flexString = (hours:number):'column'|'row' => {
    if(hours > 1){
        return 'column'
    }
    return 'row'
}

export const Entry:React.FC<EntryProps> = ({hours,name})=>{
    return <div className="timesheet-entry" style={{height:`${23*hours}px`,flexDirection:flexString(hours)}} >
        <div className="timesheet-entry-name"><span>{name}</span></div>
        <div className="timesheet-entry-hours"><span>{`${hours} Hour${hours > 1 ? 's' : '  '}`}</span></div>
    </div>
}