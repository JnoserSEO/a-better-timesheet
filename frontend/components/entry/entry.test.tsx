import React from 'react'
import { shallow } from 'enzyme'
import { Entry } from './Entry'



test('should render Entry with 1 hour', () => {
    const wrapper = shallow(<Entry hours={1} name={'Comp A'} />)
    expect(wrapper.find).toMatchSnapshot()
})

test('should render Entry with 2 hours', () => {
    const wrapper = shallow(<Entry hours={2} name={'Comp A'} />)
    expect(wrapper.find).toMatchSnapshot()
})

test('should render Entry with 4 hours', () => {
    const wrapper = shallow(<Entry hours={8} name={'Comp A'} />)
    expect(wrapper.find).toMatchSnapshot()
})


