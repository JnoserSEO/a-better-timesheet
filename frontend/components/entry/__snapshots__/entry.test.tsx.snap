// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Entry with 1 hour 1`] = `[Function]`;

exports[`should render Entry with 2 hours 1`] = `[Function]`;

exports[`should render Entry with 4 hours 1`] = `[Function]`;
